app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java

```diff
@@ -315,15 +312,21 @@ private void onFetchTokensCompletable()
      */
     private void updateTokenBalances()
     {
-        unknownAddresses.addAll(ethereumNetworkRepository.getAllKnownContracts(tokensService.getNetworkFilters()));
-        checkUnknownAddresses();
+        AddUnresolvedContracts();
         if (balanceTimerDisposable == null || balanceTimerDisposable.isDisposed())
         {
             balanceTimerDisposable = Observable.interval(0, BALANCE_CHECK_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
                     .doOnNext(l -> checkBalances()).subscribe();
         }
     }
 
+    private void AddUnresolvedContracts()
+    {
+        Observable.fromArray(ethereumNetworkRepository.getAllKnownContracts(tokensService.getNetworkFilters()).toArray(new ContractResult[0]))
+                .filter(result -> tokensService.getToken(result.chainId, result.name) == null)
+                .forEach(r -> unknownAddresses.add(r)).isDisposed();
```

Comment:
Is the `isDisposed()` at the end necessary?

Reply:
Nicely spotted. It's not necessary, but it's a harmless way to stop Java lint from complaining that the return is not used - which we never need to use in these cases.